integ-test/src/test/java/org/opensearch/sql/sql/SQLCorrectnessIT.java

```diff
@@ -32,32 +35,41 @@ protected void init() throws Exception {
   public void runAllTests() throws Exception {
     verifyQueries(EXPR_TEST_DIR, expr -> "SELECT " + expr);
     verifyQueries(QUERY_TEST_DIR, Function.identity());
+    runH2Tests();
+  }
+
+  private void runH2Tests() throws Exception {
```

Comment:
It'd be more clear if instead of ignoreDatabase list, verify took connections to test against. Then `runH2Tests` would be `verifyQueries(H2_ONLY_TEST_DIR, ..., List.of("H2"))`.

In addition, when more databases are added, this code will not break.